packages/Textarea/src/Textarea.js

```diff
@@ -30,75 +36,83 @@ const defaultProps = {
   value: "",
 };
 
-class Textarea extends React.Component {
-  componentDidMount() {
-    if (this.props.canExpand) {
-      this.resize();
-      window.addEventListener("resize", this.resize);
-    }
-  }
+function Textarea(props) {
+  let textarea = null;
 
-  componentDidUpdate(prevProps) {
-    if (this.props.canExpand && this.props.value !== prevProps.value) {
-      this.resize();
+  const resize = () => {
+    if (textarea && textarea.style) {
+      textarea.style.height = 0;
+      textarea.style.height = `${textarea.scrollHeight + 2}px`;
     }
-  }
+  };
 
-  componentWillUnmount() {
-    window.removeEventListener("resize", this.resize);
-  }
+  React.useEffect(() => {
+    if (props.canExpand) {
+      resize();
+      window.addEventListener("resize", resize);
+    }
 
-  setRef = node => {
-    this.textarea = node;
-    this.props.inputRef(node);
-  };
+    return function cleanup() {
+      window.removeEventListener("resize", resize);
+    };
+  });
 
-  resize = () => {
-    if (this.textarea && this.textarea.style) {
-      this.textarea.style.height = 0;
-      this.textarea.style.height = `${this.textarea.scrollHeight + 2}px`;
+  React.useEffect(() => {
+    if (props.canExpand) {
+      resize();
     }
+  }, [props.value]);
+
+  const setRef = node => {
+    textarea = node;
+    props.inputRef(node);
   };
 
-  render() {
-    const {
-      a11yText,
-      className,
-      canExpand,
-      hasError,
-      inputRef,
-      isDisabled,
-      isReadOnly,
-      maxHeight,
-      size,
-      ...moreProps
-    } = this.props;
-
-    if (a11yText) moreProps["aria-label"] = a11yText;
-
-    const rootClasses = classNames(
-      "form-textarea",
-      `form-textarea--${size}`,
-      { "form-textarea--is-disabled": isDisabled },
-      { "form-textarea--is-readonly": isReadOnly },
-      { "form-textarea--has-error": hasError },
-      className
-    );
-
-    return (
-      <div className={rootClasses} css={textareaStyles}>
-        <textarea
-          className="form-textarea__textarea"
-          data-pka-anchor="textarea"
-          disabled={isDisabled}
-          readOnly={isReadOnly}
-          ref={this.setRef}
-          style={{ maxHeight }}
-          {...moreProps}
-        />
-      </div>
-    );
+  const {
+    a11yText,
+    className,
+    canExpand,
+    hasError,
+    inputRef,
+    isDisabled,
+    isReadOnly,
+    maxHeight,
+    size,
+    ...moreProps
+  } = props;
+
+  if (moreProps.defaultValue) {
```

Comment:
Here, I am ensuring that only either a `defaultValue` or `value` can be set on the textarea component at any one time, prevents React warnings, as well as ensures `defaultValue` works correctly (pre this pr it was still applying `value=null` to the component even if `defaultValue` is present.